### PR TITLE
[java] type casting and numeric improvements

### PR DESCRIPTION
**Pull request by iampopovich**
### **User description**
**Thanks for contributing to Selenium!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.

### Description

While reviewing the source code, I noticed that some sections contain unnecessary type conversions in arithmetic operations and string formatting.

### Motivation and Context

I made part of these changes with the guidance of IntelliJ IDEA's code inspection. I believe that making such changes will help maintain it in a cleaner and more readable format for future maintenance.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

___

### **PR Type**
enhancement

___

### **Description**
- Refactored conditional statements to switch-case in `CdpClientGenerator.java` for cleaner code.
- Removed redundant `toString` calls and simplified exception messages in several test classes.
- Optimized hash code calculation in `Color.java` using `Long.hashCode`.
- Eliminated unnecessary division by 1 in mouse and pointer movement calculations across multiple test classes.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CdpClientGenerator.java</strong><dd><code>Refactor Conditional Statements to Switch-Case in toJson Method</code></dd></summary>
<hr>
      
java/src/org/openqa/selenium/devtools/CdpClientGenerator.java

<li>Replaced conditional statements with a switch-case structure for type <br>handling in <code>toJson</code> method.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-0e04b15ae75d735e7d1ee8a06f4691964a210f2d429a72a777e17065e5ec793e">+17/-12</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>AbstractFindByBuilder.java</strong><dd><code>Simplify Exception Message Formatting in assertValidFindBy</code></dd></summary>
<hr>
      
java/src/org/openqa/selenium/support/AbstractFindByBuilder.java

<li>Removed redundant <code>toString</code> method call in exception message <br>formatting.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-a37005b6659278ca8ee84ed612f0f0a6ecc5dabed6cc5c072784afb7b8d50645">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Color.java</strong><dd><code>Optimize Hash Code Calculation in Color Class</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
java/src/org/openqa/selenium/support/Color.java

<li>Replaced manual bitwise operation with <code>Long.hashCode</code> for hash code <br>calculation.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-dfb0df30134654347360c53aa1bae9b6f67247730ea72d400270f932d8464930">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DefaultMouseTest.java</strong><dd><code>Remove Redundant Division in Mouse Movement Test</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
java/test/org/openqa/selenium/bidi/input/DefaultMouseTest.java

- Removed unnecessary division by 1 in mouse movement coordinates.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-73693f52d08caad34072a0088cf7e5e3d6ad8809138e1b1a655c3f67c624ddc2">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DragAndDropTest.java</strong><dd><code>Simplify Exception Message in DragAndDropTest</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
java/test/org/openqa/selenium/bidi/input/DragAndDropTest.java

<li>Simplified exception message by removing redundant <code>toString</code> method <br>call.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-90c0c33bc3197e19668e269d8a92517b75ad8c1801f9ee0e68099a19bf0ba7f9">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>DefaultMouseTest.java</strong><dd><code>Clean Up Redundant Division in Mouse Interaction Test</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
java/test/org/openqa/selenium/interactions/DefaultMouseTest.java

- Removed unnecessary division by 1 in mouse movement coordinates.

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-2286706b02531fd3c0f3d217b9d4c77f24788366a2c14ab6c183c68b99ac2d76">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>PenPointerTest.java</strong><dd><code>Refactor Pointer Movement Calculations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
java/test/org/openqa/selenium/interactions/PenPointerTest.java

<li>Removed unnecessary division by 1 in pointer movement coordinates.<br> <li> Simplified calculation of center coordinates without casting.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/13909/files#diff-5c76841c952c66827ad2ba834af670e491c18973314132b87b79be7d34e49601">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: java/test/org/openqa/selenium/interactions/PenPointerTest.java [hunk 1]
```diff
@@ -407,8 +407,8 @@ public void setPointerEventProperties() {
 
     Rectangle rect = pointerArea.getRect();
 
-    int centerX = (int) Math.floor(rect.width / 2 + rect.getX());
```
Comment (iampopovich) [inline java/test/org/openqa/selenium/interactions/PenPointerTest.java:410]: in this case, I followed the rules of arithmetic operations on integers. When dividing an integer number `rect.width ` by an `int`, the result will be the integer part of the division - the same as what `Math.floor()` does, but without converting to a `double`. Therefore, the conversion to `int` after the parentheses is also unnecessary. It turns out that in this expression, only integer operands can be operated on if only the integer part of the division operation is required.

File: java/test/org/openqa/selenium/bidi/input/DefaultMouseTest.java [hunk 2]
```diff
@@ -481,7 +481,7 @@ public void testCanMoveOverAndOutOfAnElement() {
     inputModule.perform(
         windowHandle,
         getBuilder(driver)
-            .moveToElement(redbox, redSize.getWidth() / 1 + 1, redSize.getHeight() / 1 + 1)
```
Comment (iampopovich) [inline java/test/org/openqa/selenium/bidi/input/DefaultMouseTest.java:484]: here i'm waiting for `bazel test ` results on my local machine
according to the `moveToElement` signature ,` redSize.getWidth() / 1 + 1` should be `int`, `getWidth` returns `int` and dividing by 1 makes no sense
please, tell me if I'm wrong

Comment (diemol) [reply java/test/org/openqa/selenium/bidi/input/DefaultMouseTest.java:484]: Tests are passing, so it seems to be working.

**Comment by diemol** [comment]
Thank you, @iampopovich!

**Pull request merged**